src/app/[locale]/(default)/rules/[subset]/page.tsx

```diff
@@ -0,0 +1,14 @@
+import { rulesMockdata } from '@/mock-data/rules';
+import { unstable_setRequestLocale } from 'next-intl/server';
+import { notFound } from 'next/navigation';
+
+export default function RuleSubSetPage({
+  params: { subset },
+}: { params: { subset: string } }) {
+  unstable_setRequestLocale(subset);
```

Comment:
The locale needs to be set, not the subset. Change the params to also include `locale`, and use `setRequestLocale(locale)` (this does no longer have the `unstable` prefix)